src/replication/replication_node_manager.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+/*
+ * replication_node_manager.cpp
+ */
+
+#include "replication_node_manager.hpp"
+
+#include "log_impl.h"
+#include "multi_thread_stream.hpp"
+#include "replication_master_node.hpp"
+#include "replication_slave_node.hpp"
+#include "stream_file.hpp"
+
+#include <mutex>
+
+namespace cubreplication
+{
+  std::mutex commute_mtx;
+
+  std::string host_name;
+  replication_node_manager *g_instance = NULL;
+
+  replication_node_manager::replication_node_manager ()
+  {
+    INT64 buffer_size = prm_get_bigint_value (PRM_ID_REPL_CONSUMER_BUFFER_SIZE);
```

Comment:
We should unite PRM_ID_REPL_CONSUMER_BUFFER_SIZE and PRM_ID_REPL_GENERATOR_BUFFER_SIZE
into PRM_ID_REPLICATION_BUFFER_SIZE
and use the current value of PRM_ID_REPL_GENERATOR_BUFFER_SIZE (which is larger).